lib/xchain/provider/fetch.go

```diff
@@ -433,3 +422,26 @@ func headTypeFromConfLevel(conf xchain.ConfLevel) (ethclient.HeadType, bool) {
 func reqToChainVersion(req xchain.ProviderRequest) chainVersion {
 	return chainVersion{ID: req.ChainID, ConfLevel: req.ConfLevel}
 }
+
+func getLogs(ctx context.Context, rpcClient ethclient.Client, contractAddr common.Address, blockHash common.Hash, topicName string) ([]types.Log, error) {
+	metadata, err := bindings.OmniPortalMetaData.GetAbi()
+	if err != nil {
+		return nil, errors.Wrap(err, "get abi")
+	}
+
+	topics, err := abi.MakeTopics([]any{metadata.Events[topicName].ID})
```

Comment:
rather use the topic inside generated bindings, see [example](https://github.com/omni-network/omni/blob/main/halo/registry/keeper/logeventproc.go#L41)